### PR TITLE
[RFC] Use visualstudio_py_launcher in custom launcher

### PR DESCRIPTION
**Pull request by mostafaeweda**
In some development environments, complete control over starting python with a beforehand specified `pythonPath` or version is hard to attain; in addition to the use case of a dev environment that has a shell script, starting the python with the right environment and arguments setup.

For these use cases (& debugging `.par` files), the dev environment is better positioned to control enabling the debug mode when requested - here by calling `visualstudio_py_launcher.launch()`.

In my example, the setup work is done first, then:
```
# Check for a debugging environment variable (preferably set when spawning the launcher here)
import visualstudio_py_launcher
visualstudio_py_launcher.launch()
```

## Next possible improvements (waiting for comments):
* Set a `VS_PY_DEBUGGER=1` environment variable when spawning the launcher.
* Provide a `customLauncher` config that'd replace the usage of `pythonPath` and `program` -- that'd avoid piggybacking `pythonPath` & `program` to hack `customLauncher` support.

### CODE REVIEW

File: src/client/debugger/Common/Utils.ts [hunk 1]
```diff
@@ -112,6 +112,9 @@ export function getPythonExecutable(pythonPath: string): string {
 }
 
 function isValidPythonPath(pythonPath): boolean {
+    if (fs.existsSync(pythonPath)) {
+        return true;
+    }
```
Comment (mostafaeweda) [inline src/client/debugger/Common/Utils.ts]: The passed `pythonPath` is really a custom `launcher.sh` in this case, like:
```
#!/bin/bash
# do stuff
$somePythonVersion main.py
```
where `main.py` is:
```
if os.environ.get('VS_PY_DEBUGGER'):
        import visualstudio_py_launcher
        # HERE: setup `sys.argv` for the launcher (which can better be passed as arguments).
        visualstudio_py_launcher.launch()
    else:
        # run in non-debug mode
```
with debugger options:
```
{
  stopOnEntry: true,
  console: 'none',
  program: '/dev/null',
  debugOptions: ['WaitOnAbnormalExit', 'WaitOnNormalExit', 'RedirectOutput'],
  pythonPath: 'path/to/launcher.sh',
  env: {
    NUCLIDE_DEBUGGER: '1',
  },
}
```

**Comment by mostafaeweda** [comment]
cc @DonJayamanne @robertsk50

**Comment by DonJayamanne** [comment]
@mostafaeweda hi, the PTVS team is actively making changes to the Python debugger, and that's why I haven't been looking at this PR. Apologies for the radio silence

**Comment by mostafaeweda** [comment]
@DonJayamanne I appreciate your response; I'm sure you'd be excited to learn that I integrated this debugger engine for Nuclide's facebook users (& they ❤️  it ).

I've also been guilty of piggy-backing all my fixes / improvements in the PR without properly communicating all what's there.

Is there a channel that you use for communicating / collaborating on changes that I'd be able to join?

**Comment by DonJayamanne** [comment]
@mostafaeweda
For all changes to the extension, please could you submit your issues and PEs to the Microsoft repo https://github.com/Microsoft/vscode-python

We would love to have look at your improvements and interested then into the official repo for the extension.

**Pull request closed without merging**